app/models/blog.rb

```diff
@@ -10,7 +10,8 @@ class Blog < ApplicationRecord
   scope :published, -> { where('secret = FALSE') }
 
   scope :search, lambda { |term|
-    where("title LIKE '%#{term}%' OR content LIKE '%#{term}%'")
+    escaped_term = sanitize_sql_like(term.to_s)
```

Comment:
escape と sanitize で表現がズレちゃってませんかね？ https://github.com/smallmonkeykey/haunted-blog/pull/1#discussion_r2178735529 でコメントしているのは、もともと格納されていたハッシュ自体は sanitize された値ではないということです。

Reply:
すみません、やっと理解できました🙇‍♂️

https://api.rubyonrails.org/v7.1/classes/ActiveRecord/Sanitization/ClassMethods.html#method-i-sanitize_sql_like
> Sanitizes a string so that it is safe to use within an SQL LIKE statement. This method uses escape_character to escape all occurrences of itself, “_” and “%”.

で、エスケープやサニタイズという文字にひっぱられていました。
sanitize_sql_like があくまで LIKE 検索用の特殊文字をエスケープするためのメソッドであって、一般的な SQL インジェクション対策のエスケープとは別物ということがやっと理解できました。

`like_query_term`という変数名に変更しました。

Reply:
`sanitize_sql_like` の結果を受ける変数名なら `escaped` ではなく `sanitized` にする必要があるのではということをコメントしているつもりですが、それは伝わっていますか？